kiwi/tasks/system_build.py

```diff
@@ -155,11 +155,12 @@ def process(self):
         system = SystemPrepare(
             self.xml_state, image_root, True
         )
+
         manager = system.setup_repositories()
-        system.install_bootstrap(manager)
-        system.install_system(
-            manager
-        )
+        if self.xml_state.get_bootstrap_packages_sections():
+            system.install_bootstrap(manager)
+        system.install_system(manager)
+
```

Comment:
I think it would be better to handle the condition if bootstrapping should happen or not in the method install_bootstrap in system/prepare.py and not at several places in the code. I also think a warning or info message that bootstrapping will be skipped because no bootstrap packages were specified is important. We also should care for the fact that packages are not the only part of a bootstrap section:

* there could be bootstrap archives
* there could be bootstrap collections
* there could be bootstrap products

It should also be clear that bootstrap is only optional if a base image is given which contains what bootstrap normally does. So just skipping the entire step by the condition of an empty bootstrap package list is too short
thinking in my eyes.